### PR TITLE
feat/create_hotel_api

### PR DESCRIPTION
**Pull request by HaiYenNguyen12**
In this pull request, I created the following:
1. API GET a hotel by three filters (duration, city address, room type) and show details (available rooms in a hotel). 
2. API GET a hotel by Id and show details.
3. API GET a room by Id and show details.
4. API GET a booking by Id.
5. API GET all bookings of one user by email.
6. CRUD service API for admin.
7. CRUD facilities API for admin.
8.  API Get a user by email.
9.  API Get a user by Id.
10.  Reformat code.
I hope you'll be able to have time to review my code. 
Thanks.

### CODE REVIEW

File: HotelBooking.API/Controllers/HotelController.cs [hunk 1]
```diff
@@ -87,9 +87,102 @@ public async Task<IActionResult> CreateFacilityAsync([FromForm] FacilityRequest
...
         }
 
+        [Authorize(Roles = "Administrator", AuthenticationSchemes = "Bearer")]
+        [HttpPut("update-facility")]
+        public async Task<IActionResult> UpdateFacilityAsync(FacilityModel model, Guid id)
+        {
+            if (model == null)
+            {
+                logger.LogError("Invalid model sent from client.");
+                return BadRequest("Invalid model object");
```
Comment (levtmadden) [inline HotelBooking.API/Controllers/HotelController.cs:97]: I think we don't care about invalid data before triggering business logic. So this error could be redundant for me

File: HotelBooking.API/Controllers/HotelController.cs [hunk 2]
```diff
@@ -87,9 +87,102 @@ public async Task<IActionResult> CreateFacilityAsync([FromForm] FacilityRequest
...
+                StatusCode = HttpStatusCode.BadRequest,
+                IsSuccess = false
+            });
+        }
+
+        [Authorize(Roles = "Administrator", AuthenticationSchemes = "Bearer")]
+        [HttpDelete("delete-facility")]
+        public async Task<IActionResult> DeleteFacility(Guid id)
+        {
+            var result = await hotelService.DeleteFacilityAsync(id);
```
Comment (levtmadden) [inline HotelBooking.API/Controllers/HotelController.cs:123]: add check exist before deleting, should return 400 when not exist

File: HotelBooking.API/Controllers/HotelController.cs [hunk 3]
```diff
@@ -87,9 +87,102 @@ public async Task<IActionResult> CreateFacilityAsync([FromForm] FacilityRequest
...
+            {
+                return StatusCode(StatusCodes.Status200OK, new ResponseModel
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    IsSuccess = true,
+                    Data = result
+                });
+            }
+            return StatusCode(StatusCodes.Status404NotFound, new ResponseModel
+            {
```
Comment (levtmadden) [inline HotelBooking.API/Controllers/HotelController.cs:157]: you can write it in a better way
return result != null ? Ok() : NotFound()
don't need if else for this case

File: HotelBooking.Data/Extensions/MailSender.cs [hunk 4]
```diff
@@ -8,14 +7,14 @@ namespace HotelBooking.Data.Extensions;
...
+    public async Task<bool> SendInforOfBooking(string toEmail, string idBooking)
     {
-        
+
         var email = new MimeMessage();
         email.From.Add(MailboxAddress.Parse("girlmilk123@gmail.com"));
         email.To.Add(MailboxAddress.Parse(toEmail));
         email.Subject = "Confirm EMail";
-        email.Body = new TextPart(TextFormat.Plain) { Text = model.InformationSender()};
+        email.Body = new TextPart(TextFormat.Plain) { Text = $"https://localhost:7137/api/Booking/get-booking-by-id?id={idBooking}" };
```
Comment (levtmadden) [inline HotelBooking.Data/Extensions/MailSender.cs:17]: I'm not a fan of this kind. You can create a method name generateEmailBody() and update anything related to email body in this method.

File: HotelBooking.Service/Services/HotelService.cs [hunk 5]
```diff
@@ -144,5 +223,84 @@ public async Task<bool> AddServiceAndFacilityToRoomAsync(EquipRoomRequest model)
...
+
+                foreach (var hotel in result)
+                {
+                    var rooms = await roomRepository.GetByCondition(x => x.HotelId.Equals(hotel.Id) && x.RoomType == model.RoomType).ToListAsync();
+                    if (rooms.Any())
+                    {
+                        foreach (var room in rooms)
+                        {
+                            var checkResult = await checkDurationValidationService.CheckValidationDurationForRoom(model.Duration, room.Id);
+                            if (checkResult == true)
```
Comment (levtmadden) [inline HotelBooking.Service/Services/HotelService.cs:249]: to much nesty logic. Need to refactor to reduce complexity

File: HotelBooking.API/Controllers/HotelController.cs (hunk 1 above)
Comment (HaiYenNguyen12) [reply HotelBooking.API/Controllers/HotelController.cs]: I fixed it. thanks

File: HotelBooking.API/Controllers/HotelController.cs (hunk 3 above)
Comment (HaiYenNguyen12) [reply HotelBooking.API/Controllers/HotelController.cs]: I fixed it. thanks.

File: HotelBooking.Service/Services/HotelService.cs (hunk 5 above)
Comment (HaiYenNguyen12) [reply HotelBooking.Service/Services/HotelService.cs]: I fixed it. thanks.

File: HotelBooking.API/Controllers/HotelController.cs (hunk 2 above)
Comment (HaiYenNguyen12) [reply HotelBooking.API/Controllers/HotelController.cs:89]: I fixed it. thanks

File: HotelBooking.Data/Extensions/MailSender.cs (hunk 4 above)
Comment (HaiYenNguyen12) [reply HotelBooking.Data/Extensions/MailSender.cs]: I fixed it. thanks

**Pull request closed without merging**